sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_repository.py

```diff
@@ -0,0 +1,223 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+from typing import TYPE_CHECKING
+
+from azure.core.exceptions import (
+    ClientAuthenticationError,
+    ResourceNotFoundError,
+    ResourceExistsError,
+    HttpResponseError,
+    map_error,
+)
+from azure.core.paging import ItemPaged
+from azure.core.tracing.decorator import distributed_trace
+
+from ._base_client import ContainerRegistryBaseClient
+from ._generated.models import AcrErrors
+from ._helpers import _parse_next_link
+from ._models import (
+    DeleteRepositoryResult,
+    ArtifactManifestProperties,
+    RepositoryProperties,
+)
+from ._registry_artifact import RegistryArtifact
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+    from azure.core.credentials import TokenCredential
+    from ._models import ContentProperties
+
+
+class ContainerRepository(ContainerRegistryBaseClient):
```

Comment:
Why is this no longer a client? Is it because a user cannot instantiate it themselves?

Reply:
A user could instantiate it themselves but it would look a little awkward. We received feedback from the service team that the two clients made it confusing on where a user should start. Krystof and Anne Thompson came up with a design where both `ContainerRepository` and `RegistryArtifact` have client-like properties (being able to make service calls) but are more like models. This produces a more intuitive starting point with `ContainerRegistryClient`.